File: paper_artifacts/experiments/llr9/artifacts/sources/kimi27sglang-c/tsvc_2_s119__submitted.c
#include <stdint.h>

void tsvc_2_s119_fp64(double *restrict aa, double *restrict bb, int64_t LEN_2D,
                      uint8_t *restrict workspace, int64_t workspace_bytes) {
    #pragma omp parallel
    {
        for (int64_t i = 1; i < LEN_2D; ++i) {
            double *restrict aa_i = aa + i * LEN_2D;
            const double *restrict aa_im1 = aa + (i - 1) * LEN_2D;
            const double *restrict bb_i = bb + i * LEN_2D;
            #pragma omp for simd
            for (int64_t j = 1; j < LEN_2D; ++j) {
                aa_i[j] = aa_im1[j - 1] + bb_i[j];
            }
        }
    }
}

File: paper_artifacts/experiments/llr9/artifacts/sources/oss120b-fortran/tsvc_2_s119__submitted.f90
! Kernel for tsvc_2_s119 (fp64 version) - diagonal processing with accumulator for minimal memory traffic
subroutine tsvc_2_s119_fp64(aa, bb, LEN_2D, workspace, workspace_bytes) bind(C, name="tsvc_2_s119_fp64")
   use iso_c_binding
   integer(c_int64_t), value :: LEN_2D
   type(c_ptr), value :: workspace
   integer(c_int64_t), value :: workspace_bytes
   real(c_double), intent(inout) :: aa(LEN_2D, LEN_2D)
   real(c_double), intent(in) :: bb(LEN_2D, LEN_2D)
   integer :: d, i, i_start, i_end, j
   real(c_double) :: acc

   ! Parallelize over diagonals (i - j = d). Each diagonal can be computed independently.
   !$omp parallel do private(d, i, i_start, i_end, j, acc) schedule(static)
   do d = -(LEN_2D-2), LEN_2D-2
      i_start = max(2, d+2)
      i_end   = min(LEN_2D, LEN_2D + d)
      j = i_start - d
      ! Initialize accumulator with the element just before the start of this diagonal.
      acc = aa(i_start-1, j-1) + bb(i_start, j)
      aa(i_start, j) = acc
      do i = i_start+1, i_end
         j = i - d
         acc = acc + bb(i, j)
         aa(i, j) = acc
      end do
   end do
   !$omp end parallel do

   ! workspace arguments unused but required by harness
end subroutine tsvc_2_s119_fp64
